examples/main/main.cpp

```diff
@@ -47,6 +47,53 @@ void sigint_handler(int signo) {
 }
 #endif
 
+void command(std::string buffer, gpt_params params, const int n_ctx ) {
+    // check buffer's first 3 chars equal '???' to enter command mode. 
+    if (strncmp(buffer.c_str(), "???", 3) == 0 && buffer.length() > 3) {
```

Comment:
Nit: please negate this condition and do an easy return so the function body doesn't have to be indented again:
```suggestion
    if (buffer.length() < 3 || strncmp(buffer.c_str(), "???", 3) != 0) return;
```

Reply:
done mate.